app/models/patient.rb

```diff
@@ -97,6 +97,7 @@ class Patient < ApplicationRecord
   has_many :close_contacts
   has_many :contact_attempts
 
+  before_save :set_time_zone_offset
```

Comment:
I also have thought about storing the `time_zone_offset` as a value on the Patient model, but this feels like an expensive cost for doing that. Patients are updated constantly, and most of the time this will be a useless operation unless they change their address OR it's a daylight savings time switch. Have we considered the performance implications of this?